lib/mix/lib/mix/dep/loader.ex

```diff
@@ -108,8 +108,9 @@ defmodule Mix.Dep.Loader do
         make?(dep) ->
           make_dep(dep)
 
+        # TODO: test
         true ->
-          {dep, []}
+          {dep, Enum.map(children || [], &to_dep(&1, opts[:dest], manager))}
```

Comment:
I may have given bad advice but we want `nil` here. So if you have a :bare dep that depends on a rebar dep, rebar gets used.

```suggestion
          {dep, Enum.map(children || [], &to_dep(&1, opts[:dest], _manager = nil))}
```